Update deny list to all system packages for ARMv6 HF.

Alexhuszagh opened the pull request:
Currently, when using `arm-unknown-linux-gnueabihf`, users frequently try to install packages for the `armhf` Debian architecture, objects which are compiled for ARMv7-A, and therefore are not compatibility with ARMv6 platforms such as the Raspberry Pi 0. This differs from the soft-float targets, where packages for the `armel` architecture are compiled for the ARMv5TE architecture, and therefore are compatible with ARMv5TE, ARMv6, and ARMv7 soft-float targets.

Deny-listing all packages for the `armhf` architecture prevents issues such as in #1017, where the use of ARMv7 system packages are linked to the binary, so the resulting binary can no longer be executed on an ARMv6 platform. This is difficult to detect until the user attempts to use the generated objects (binaries, libraries) on native hardware.

Also linked is #426.

Alexhuszagh commented:
Marking as a draft since I haven't tested this logic extensively, and denylists an entire architecture, so it's likely a good idea to be careful. Also, we might want to do this for a few musl targets in case if installing system packages can affect their dynamic linking (since this has been an issue with x86_64, see #902).

tmayoff commented:
This seems like a decent solution, as long as it's mentioned in docs somewhere (in case I forget about this and try installing a package that I know exists but it can't find), also I'm not sure if it is mentioned somewhere but a note that at least the `:armhf` packages are built for armv7 and that for armv6 might need to be compiled from source would also be good.

Alexhuszagh commented:
> This seems like a decent solution, as long as it's mentioned in docs somewhere (in case I forget about this and try installing a package that I know exists but it can't find), also I'm not sure if it is mentioned somewhere but a note that at least the `:armhf` packages are built for armv7 and that for armv6 might need to be compiled from source would also be good.

This needs to be added to the Wiki then as well, under the FAQ section then:
https://github.com/cross-rs/cross/wiki/FAQ

I'll do that later, thank you.

Emilgardis commented:
we don't have mentions for the other pins, probably a good idea to document it

Alexhuszagh commented:
I've added documentation on the pinned packages [here](https://github.com/cross-rs/cross/wiki/FAQ#cannot-find-package), and where we use them. I added the `arm-unknown-linux-gnueabihf` example as well since I'm assuming this will be merged: if not, we should still document this lack of compatibility somewhere.

Alexhuszagh commented:
The musl images don't need to deny the Debian system packages as expected since the musl sysroot is different, and the compiler doesn't try to link to those packages.

Emilgardis reviewed (approved):
awesome!

bors r+

Alexhuszagh commented:
bors r=Emilgardis

Emilgardis commented:
haha nice catch on the wrong commit message 😅
